src/app/src/main/java/com/example/bookmark/BorrowedActivity.java

```diff
@@ -78,22 +79,48 @@ protected boolean getBookStatusVisibility() {
      */
     @Override
     protected void getBooks() {
-        StorageServiceProvider.getStorageService().retrieveBooksByRequester(
-                user,
-                books -> {
-                    visibleBooks.clear();
-                    relevantBooks.clear();
-                    for (Book book : books) {
-                        if (book.getStatus() == Book.Status.BORROWED) {
-                            relevantBooks.add(book);
+        if (user == null) {
+            String username = UserUtil.getLoggedInUser(this);
+            StorageServiceProvider.getStorageService().retrieveUserByUsername(username,
+                    user1 -> {
+                        StorageServiceProvider.getStorageService().retrieveBooksByRequester(
+                                user1,
+                                books -> {
+                                    visibleBooks.clear();
+                                    relevantBooks.clear();
+                                    for (Book book : books) {
+                                        if (book.getStatus() == Book.Status.BORROWED) {
+                                            relevantBooks.add(book);
+                                        }
+                                    }
+                                    visibleBooks.addAll(relevantBooks);
+                                    visibleBooksAdapter.notifyDataSetChanged();
+                                }, e -> {
+                                    DialogUtil.showErrorDialog(this, e);
+                                }
+                        );
+                    }, e -> {
+                        DialogUtil.showErrorDialog(this, e);
+                    }
+            );
+        } else {
+            StorageServiceProvider.getStorageService().retrieveBooksByRequester(
```

Comment:
May want to refactor this to a separate function to reduce duplication of code between the `if` clause and this `else` clause.

Reply:
Good thinking